packages/tkg-clusterclass-vsphere/bundle/config/upstream/base.yaml

```diff
@@ -892,13 +885,15 @@ spec:
           template: |
             {{ $first := true }}
             {{- range $key, $val := (index .TKR_DATA .builtin.controlPlane.version).labels }}
+            {{- if regexMatch "^(?:[a-zA-z])(?:[-\\w\\.]*[a-zA-z])$" $val }}
```

Comment:
Hi Nick, I'm trying to understand the code, this part of jsonPath has nothing to do with the custom Node Labels feature https://docs.vmware.com/en/VMware-Tanzu-Kubernetes-Grid/1.5/vmware-tanzu-kubernetes-grid-15/GUID-tanzu-k8s-clusters-config-plans.html#custom-node-labels-8 , right? It's specifically for Node Pool.

Reply:
It is both. This allows customers to define custom node labels and they can do it per node pool.